lib/binary_to_decimal.rb

```diff
@@ -5,5 +5,10 @@
 # Calculate  and return the decimal value for this binary number using
 # the algorithm you devised in class.
 def binary_to_decimal(binary_array)
-  raise NotImplementedError
+  i = 0
+  decimal = 0
+  binary_array.length.times do |i|
+    decimal += binary_array[i] * 2 ** (7 - i)
+  end
```

Comment:
Overall, I think your code looks great! It's very concise! One comment that I have is on line 11, in the case where you would want to scale this program to intake binary number greater than 8 numbers, you might want to modify exponent of the 2 (the 7-i part). Perhaps setting the binary_array.length = n and the exponent would be (n-i). For example if the binary_array.length = 9 then the exponent would be (9-i) Let me know if this does not make sense :)

Reply:
That's great feedback! To address this feedback, how about leveraging the length of the array? Instead of `7`, use `binary_array.length - 1`.